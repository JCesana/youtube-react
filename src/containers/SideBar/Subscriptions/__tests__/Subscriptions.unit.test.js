import React from 'react';
import {shallow} from 'enzyme';
import Subscriptions from '../Subscriptions';

test('renders', () => {
  const wrapper = shallow(
    <Subscriptions/>
  );
  expect(wrapper).toMatchSnapshot();
});
