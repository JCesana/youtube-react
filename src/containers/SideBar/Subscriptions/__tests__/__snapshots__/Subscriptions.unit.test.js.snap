// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription renders broadcasting subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Productioncoder
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`Subscription renders empty subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;

exports[`Subscription renders non-broadcasting subscription with new videos 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Productioncoder
      </span>
    </div>
    <span
      className="new-videos-count"
    >
      4
    </span>
  </div>
</MenuItem>
`;

exports[`Subscriptions renders 1`] = `
<Fragment>
  <SideBarHeader
    title="Subscriptions"
  />
  <Subscription
    broadcasting={true}
    label="MusicChannel"
  />
  <Subscription
    amountNewVideos={10}
    label="Coursea"
  />
  <Subscription
    amountNewVideos={23}
    label="TEDx Talks"
  />
  <Subscription
    amountNewVideos={4}
    label="Stanford iOS"
  />
  <Subscription
    amountNewVideos={114}
    label="Udacity"
  />
  <Divider />
</Fragment>
`;

exports[`renders 1`] = `
<Fragment>
  <Component
    title="Subscriptions"
  />
  <Subscription
    broadcasting={true}
    label="MusicChannel"
  />
  <Subscription
    amountNewVideos={10}
    label="Coursea"
  />
  <Subscription
    amountNewVideos={23}
    label="TEDx Talks"
  />
  <Subscription
    amountNewVideos={4}
    label="Stanford iOS"
  />
  <Subscription
    amountNewVideos={114}
    label="Udacity"
  />
  <Divider />
</Fragment>
`;

exports[`renders Subscriptions 1`] = `
<Fragment>
  <Component
    title="Subscriptions"
  />
  <Subscription
    broadcasting={true}
    label="MusicChannel"
  />
  <Subscription
    amountNewVideos={10}
    label="Coursea"
  />
  <Subscription
    amountNewVideos={23}
    label="TEDx Talks"
  />
  <Subscription
    amountNewVideos={4}
    label="Stanford iOS"
  />
  <Subscription
    amountNewVideos={114}
    label="Udacity"
  />
  <Divider />
</Fragment>
`;
